src/NfcIpcSocket.cpp

```diff
@@ -90,34 +90,88 @@ int NfcIpcSocket::getListenSocket() {
   return nfcdConn;
 }
 
+int NfcIpcSocket::getConnectedSocket(const char* socketName)
+{
+  size_t len = strlen(socketName);
+  if (len > (SIZE_MAX - 2)) {
+    ALOGE("Socket address too long\n");
+    return -1;
+  }
+
+  size_t siz = len + 2;
+  if (siz > UNIX_PATH_MAX) {
+    ALOGE("Socket address too long\n");
+    return -1;
+  }
+
+  struct sockaddr_un addr;
+  addr.sun_family = AF_UNIX;
+  addr.sun_path[0] = '\0'; /* abstract socket namespace */
+  memcpy(addr.sun_path + 1, socketName, len + 1);
+  socklen_t addrLen = offsetof(struct sockaddr_un, sun_path) + siz;
+
+  int nfcdRw = socket(AF_UNIX, SOCK_STREAM, 0);
```

Comment:
There is also a global static int nfcdRw variable in NfcIpcSocket.cpp, could you help either not declare nfcdRw here or remove the global static variable.